File: client/src/app/shared/models/balance-sheet.model.ts
import IBalanceSheetAccountInfo from "./balance-sheet-account-info.model";

export default interface IBalanceSheet {
  assetsAmount: number;
  liabilitiesAmount: number;
  equityAmount: number;
  assetsAccountsInfoList: IBalanceSheetAccountInfo[];
  liabilitiesAccountsInfoList: IBalanceSheetAccountInfo[];
  equityAccountsInfoList: IBalanceSheetAccountInfo[];
  fromDate: string;
  toDate: string;
  netIncome: number
}

File: API/Utility/BalanceSheetModel.cs
namespace API.Utility;

public class BalanceSheetModel
{
    public decimal AssetsAmount { get; set; }
    public decimal LiabilitiesAmount { get; set; }
    public decimal EquityAmount { get; set; }
    
    public List<AccountInfo> AssetsAccountsInfoList { get; set; } = null!;
    public List<AccountInfo> LiabilitiesAccountsInfoList { get; set; } = null!;
    public List<AccountInfo> EquityAccountsInfoList { get; set; } = null!;
    
    public string FromDate { get; set; } = null!;
    public string ToDate { get; set; } = null!;

    public decimal NetIncome { get; set; }
}